Show which type was the wrong one when a signature is invalid

smessmer opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #39611 Unify TensorOptions signatures
* #39492 Move TensorOptions ops to c10
* #38361 Assert that kernels are called with the right signature
* #39823 TORCH_FN
* **#39491 Show which type was the wrong one when a signature is invalid**

-

Differential Revision: [D21872519](https://our.internmc.facebook.com/intern/diff/D21872519/)

smessmer committed:
Show which type was the wrong one when a signature is invalid

-

Differential Revision: [D21872519](https://our.internmc.facebook.com/intern/diff/D21872519/)

[ghstack-poisoned]

bhosmer commented on aten/src/ATen/core/boxing/impl/make_boxed_from_unboxed_functor.h:
```diff
@@ -51,7 +51,7 @@ using supported_primitive_arg_types = guts::typelist::typelist<
       }, /* else */ [] {
         auto tmap = c10::getCustomClassTypeMap();
-        TORCH_CHECK(c10::isCustomClassRegistered<T>(), "Tried to use undefined class as input argument");
+        TORCH_CHECK(c10::isCustomClassRegistered<T>(), "Tried to use undefined class ", c10::util::get_fully_qualified_type_name<T>(), " as input argument");
```
Don't know how you guys feel about this but I've been linewrapping this kind of stuff.

smessmer committed:
Update on "Show which type was the wrong one when a signature is invalid"


-

Differential Revision: [D21872519](https://our.internmc.facebook.com/intern/diff/D21872519/)

[ghstack-poisoned]

smessmer committed:
Update on "Show which type was the wrong one when a signature is invalid"


-

Differential Revision: [D21872519](https://our.internmc.facebook.com/intern/diff/D21872519/)

[ghstack-poisoned]

smessmer committed:
Update on "Show which type was the wrong one when a signature is invalid"


-

Differential Revision: [D21872519](https://our.internmc.facebook.com/intern/diff/D21872519/)

[ghstack-poisoned]
